code/orders-api-rest/src/main/java/com/academy/orders/apirest/common/ErrorHandler.java

```diff
@@ -75,4 +77,20 @@ public ErrorObjectDTO handleBadRequestException(final MethodArgumentTypeMismatch
 				.title(HttpStatus.BAD_REQUEST.getReasonPhrase()).detail(error.getMessage());
 	}
 
+	@ExceptionHandler(EmptyCartException.class)
+	@ResponseStatus(value = HttpStatus.BAD_REQUEST)
+	public ErrorObjectDTO handleEmptyCartException(final EmptyCartException ex) {
+		log.warn("Cart is empty.", ex);
+		return new ErrorObjectDTO().status(HttpStatus.BAD_REQUEST.value())
+				.title(HttpStatus.BAD_REQUEST.getReasonPhrase()).detail(ex.getMessage());
+	}
+
+	@ExceptionHandler(InsufficientProductQuantityException.class)
+	@ResponseStatus(value = HttpStatus.BAD_REQUEST)
```

Comment:
Maybe better to use different status, for example CONFLICT

Reply:
Improved